src/aggregators.ts

```diff
@@ -0,0 +1,110 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Red Hat, Dharmendra Patel 2020
+ * Licensed under the Apache-2.0 License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+'use strict';
+import { Package } from './package';
+import compareVersions = require('compare-versions');
+
+/* PackageAggregator */
+class PackageAggregator {
+    ecosystem: string;
+    packages: Array<Package> = Array<Package>();
+    isNewPackage: boolean;
+
+    constructor(ecosystem: string) {
+        this.ecosystem = ecosystem;
+    }
+
+    aggregate(newPackage: Package): Package {
+        return null;
+    }
+}
+
+/* Noop Package aggregator class */
+class NoopPackageAggregator extends PackageAggregator {
+    constructor() {
+        super("");
+    }
+
+    aggregate(newPackage: Package): Package {
+        // Make it a new package always and set ecosystem for package.
+        this.isNewPackage = true;
+        newPackage.ecosystem = this.ecosystem;
+
+        return newPackage;
+    }
+}
+
+/* Golang Package aggregator class */
+class GolangPackageAggregator extends PackageAggregator {
+    constructor() {
+        super("golang");
+    }
+
+    aggregate(newPackage: Package): Package {
+        // Set ecosystem for new package from aggregator
+        newPackage.ecosystem = this.ecosystem;
```

Comment:
Why this ecosystem attribute should go to base and used again used only from the derived class? Why not directly use it from derived?
```suggestion
    }

    aggregate(newPackage: Package): Package {
        // Set ecosystem for new package from aggregator
        newPackage.ecosystem = "golang";
```

Reply:
I will modify code slightly to completely avoid ecosystem attribute.

Reply:
Accepted and done.